frontend/src/app/_services/route.service.ts

```diff
@@ -0,0 +1,285 @@
+import {Injectable} from '@angular/core';
+import {BackendPath, FrontendPath, NavigationParams, NavigationRoute} from '../_models/route';
+import {MenuHierarchy} from '../_models/menu';
+
+const EMPTY_CHAR = '_';
+const EMPTY_CHAR_REGEX = /_/g;
+export const NAVIGATE_PATH = 'app/documentation/navigate/';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class RouteService {
+
+    constructor() {
+    }
+
+    static legacyFullFrontEndUrlToFullFrontEndUrl(input: string): string {
```

Comment:
I don't think we should use static methods. The service should be injected in components that need it, and instance methods can be used.

Reply:
I kwnow, I though we couldn't in a Pipe, but in fact we can.Thanks